[Andy Chong Weikang] iP

andychong23 opened the pull request:
# Bob
> "They say you never know until you try." - Anonymous

If you are hesitant about trying this product because it is going to be like any other chatbot, read the above quote and try it out first.


## What does Bob do
Bob frees your mind of having to remember things you need to do. It is:
* text-based
* easy to use

## Features
- [x] Managing Tasks
- [x] Finding Tasks
- [ ] Managing deadlines (coming soon!)
- [ ] Reminders (coming soon!)

## How to use this chatbot :smiley:
1. Download the latest release [here](https://github.com/andychong23/ip/releases)
2. **Navigate to the folder** that you have downloaded the release to
3. Run the code below on your **__terminal__**
```
java -jar bob.jar
```
4. You can then add your tasks
5. Let it manage all your tasks for you 
6. Free your mind to do whatever you want to do

andychong23 committed:
Move Duke.java into its own package

andychong23 committed:
Include package into Duke.java

andychong23 committed:
Add io package and the classes that are associated with it

andychong23 committed:
Change BufferedReader attribute to private

andychong23 committed:
Write first iteration of Output Class

andychong23 committed:
Modify .gitignore to ignore .class extensions

andychong23 committed:
Change scope of bufferedWriter to final

andychong23 committed:
Change scope of bufferedReader to final

andychong23 committed:
Revert "Change scope of bufferedWriter to final"

This reverts commit 76325976b3c0a1e1207b066d16d14f5c9f22317e.

andychong23 committed:
Change scope of bufferedReader to final

andychong23 committed:
Modify Output.java

Modify internal Implementation of initialization of BufferedWriter and added end arguement to printOutputMethod

andychong23 committed:
Modify elements access in for loop to include index

andychong23 committed:
Implement methods to display welcome and exit messages

andychong23 committed:
Rename name of chatbot to Bob

andychong23 committed:
Remove .class files

andychong23 committed:
Add cleanUserInput method

Wrap cleanUserInput method around getUserInput

andychong23 committed:
Add new methods

New methods added are: getUserInput, displayMessage. Methods modified: displayWelcomeMessage

andychong23 committed:
Add header comment to methods

andychong23 committed:
Change file name from Duke to Bob

andychong23 committed:
Add ability for user to enter user input

andychong23 committed:
Add tasks to be done in Bob for Level-2

andychong23 committed:
Add displayMessage as a new method to UI

andychong23 committed:
Created abstract base class of Task

andychong23 committed:
Add subclasses of Task

ReadTask and ReturnTask is created

andychong23 committed:
Add first draft of EventHandler

andychong23 committed:
Add First draft of User

andychong23 committed:
Change access modifiers for getTaskName and getTaskDetail

andychong23 committed:
Format string in getTaskList method of User class

andychong23 committed:
Change input validation from Input class to UI class

andychong23 committed:
Implemented processEvent method code

Add header comments to class

andychong23 committed:
Remove declaration of taskName

taskName attrtibute is already defined in the Task abstract class

andychong23 committed:
Implemented code to get the String representation fo the taskList

andychong23 committed:
Introduced User and EventHandler objects

Move event processing to the EventHandler class

andychong23 committed:
Move Task class and subclasses to their own package

Renamed the packages in the other java files

andychong23 committed:
Remove ReadTask.java and ReturnTask.java

These should all be considered as a task and should not have a specific class to handle each scenario

andychong23 committed:
Modified EventHandler class

Removed how tasks are defined. Instead of defining each task type, encapsulate under a Task class

andychong23 committed:
Refactor Task class to not be an abstract class to generalise it

Implemented the required methods for its functionality

andychong23 committed:
Change the formatted string to match what is required

andychong23 committed:
Implemented attribute to check if Task is done

Implemented getters and setters to access the attribute

andychong23 committed:
Refined header comment for getUserInput method

List that the returned String input has not trailing or leading whitespaces

andychong23 committed:
Implemented isInputValid method

Covers list,bye,mark,unmark for input validation

andychong23 committed:
Add markTaskAsDone, markTaskAsNotDone, getMarkTaskConsoleOutput and modified getTaskList

markTaskAsDone and markTaskAsNotDone changes the status of the Task number indicated.

getMarkTaskConsoleOutput returns the output message required to be displayed to the user

getTaskList is changed to include the task status

andychong23 committed:
Add enum for event types and implemented processEvent method

processEvent method checks the user input to the event type specified and carries out the appropriate actions

andychong23 committed:
Change EventHandler to ActionHandler

This is to reduce confusion between the Event task and the actual event to carry out processing for

andychong23 committed:
Change enum class from Event to Action in ActionHandler

Updated UI.java and Task.java to reflect changes accordingly

andychong23 committed:
Add new types of tasks that extends Task

andychong23 committed:
Add first draft of providing input validation for TODO, DEADLINE and EVENT actions

andychong23 committed:
Input validation for TODO, DEADLINE and EVENT is completed

andychong23 committed:
Implement HasDeadline Interface

andychong23 committed:
Implement HasStart Interface

andychong23 committed:
Implement the code for DeadLineTask class

andychong23 committed:
Implement the code for EventTask class

andychong23 committed:
Overloaded addTask method, implemented getTaskInformation method and implemented getNumberOfTasks

New addTask method returns the index of the task added in this list

getTaskInformation method returns the formatting of what is to be displayed to the user on the console

getNumberOfTasks returns the size of the current task list

andychong23 committed:
Implemented handling of TODO, DEADLINE and EVENT actions

Converted eventStringTokens to be implemented as a List data structure instead of an array

Overloaded the createTask method to handle TODO, DEADLINE and EVENT actions

andychong23 committed:
Implemented input validation

andychong23 committed:
Remove comments to complete Level-4

andychong23 committed:
Modify getStartDateTime to get the correct attribute

andychong23 committed:
Set up automated testing

andychong23 committed:
Shift unit test to corefunctionality folder

andychong23 committed:
Create test to test for error messages returned to the user

andychong23 committed:
Added ValidationToken class to encapsulate error message into the validation token

This helps to maintain the functionality of the isInputValid method in UI class

This also standardizes the input validation through the use of the ValidationToken

andychong23 committed:
Convert the return objects for isInputValid method from boolean to ValidationToken

andychong23 committed:
Change runtest.sh to loop through the test directories

andychong23 committed:
Add delete as a new action

DELETE is added to the Action enum class and processEvent handles delete input command now
Remove default handling of input that does not match any action in the Action enum class

andychong23 committed:
Add inputErrors for delete command and provide better name for incorrect argument types

Incorrect argument type is changed for mark command from mark_incorrect_argument to mark_incorrect_argument_type

Incorrect argument type is changed for unmark command from unmark_incorrect_argument to unmark_incorrect_argument_type

andychong23 committed:
Added new method deleteTask for User class

Modified ActionHandler to change method name from removeTask to deleteTask for consistency with action name

andychong23 committed:
Implement input validation for delete command and modify InputError enum for mark and unmark commands

InputError enum for mark is changed from MARK_INCORRECT_ARGUMENT to MARK_INCORRECT_ARGUMENT_TYPE
InputError enum for unmark is changed from UNMARK_INCORRECT_ARGUMENT to UNMARK_INCORRECT_ARGUMENT_TYPE

andychong23 committed:
Rearrange imports

andychong23 committed:
Added header comments to all classes and methods

andychong23 committed:
Refactor string building to use String.join instead of StringBuilder

andychong23 committed:
Include INVALID_TASK_NUMBER as a new input error

This covers cases where user has to pass in a task number that does not exist in the list

andychong23 committed:
Modify isInputValid method to receive User as a new parameter

This allows task number input validation to be done to ensure that the task number exists

andychong23 committed:
Pass in User to ui.getValidUserInput

andychong23 committed:
Change UI variable to uI to follow code standards

andychong23 committed:
Add code to handle data storage capabilities

Add a Path object to User class

Add DataHandler class to handle all tasks relating reading and writing data

andychong23 committed:
Add initial implementation of DataHandler

Methods such as writeFile and readFile is implemented

andychong23 committed:
Modified User class to be able to save user data

Implemented getter for dataFilePath

Implemented getSaveData which returns the save information for the tasks

Implmented a new constructor that takes in the userName to look for the correct saved file

andychong23 committed:
Implemented getSaveInformation method to return task to be saved

andychong23 committed:
Override superclass getSaveInformation method

andychong23 committed:
Modified writeFile and saveFile implementation to handle missing files

Include saveDelimeter as a global variable to be accessed

andychong23 committed:
Implemented functionality to save user data in ActionHandler

andychong23 committed:
Add default user name to the implementation

andychong23 committed:
Rename delemiter to delimiter

andychong23 committed:
Change createTask method to become a static method

This change ensures that users can recreate the task list from their save file

Added a DEFAULT action to handle all switch cases

andychong23 committed:
Add programRoot static variable

andychong23 committed:
Implemented the construction of the full file path to get the save file

Created the first draft to build the task list from the save file

andychong23 committed:
Implemented mapTaskType method that returns the taskType

This is done to allow the User class to recreate the task list from the save file

andychong23 committed:
Change default username to bob

andychong23 committed:
Change how writeFile and readFile is implemented

writeFile method: There is no need to check if file exists as the start up of the programme always tries to read the file

If the file does not exist, the file will be created at the start

readFile method: Change the way file creation is done if the file does not exists

andychong23 committed:
Modify implementation of constructor, buildTaskList and changed getSaveData to createSaveData

Constructor now builds the task list when the user gets initialised with a user name

buildTaskList splits the saveData correctly. Fixed the issue where the saveData was not split on saveDelimiter due to regex evaluation

buildTaskList builds the taskDetails format accordingly so that ActionHandler.createTask can be called

buildTaskList handles scenarios where ActionHandler.createTask returns null

Changed method from getSaveData to createSaveData to make the method clearer on what it is actually doing

andychong23 committed:
Changed method from getSaveInformation to createSaveData

This change better reflects what the method is supposed to do

Changed saveInformation payload for EventTask and DeadLineTask

andychong23 committed:
Change the method call for Task objects to create their save data

andychong23 committed:
Reflect method name changes from User class

andychong23 committed:
Refactor getTaskInformation method from User class to Task class

andychong23 committed:
Modify runtest.sh for correct program functionality with files

andychong23 committed:
Modify getTaskInformation implementation after refactoring from User to Task class

andychong23 committed:
Remove todo comment

andychong23 committed:
Modify dataFilePath such that it reflects the correct file paths to find the storage file

Modified methods to getTaskInformation from the task instead of calling the getTaskInformation method from the User class

andychong23 committed:
Merge branch 'branch-Level-7' into master

Implemented Level-7 of the ip

Main implementations:
1. Implemented save feature such that users can save their event logs
2. Refactored getTaskInformation from User to Task class
3. Modified runtest.sh such that user logs are found in the correct location

andychong23 committed:
Refactor date and datetime validation to InputValidation class from UI class

andychong23 committed:
Implement parsing of date/datetime string to date/datetime objects

Implemented superclass method to parse dates and datetimes

Modified constructors and getters in both EventTask and DeadLineTask to accomodate date/datetime objects

andychong23 committed:
Change DateTimeFormatter format for isValidDate

andychong23 committed:
Change date separator in parseDateTime method

andychong23 committed:
Refactor input validation code to InputValidator class from UI class

This was done to abstract away implementation from the UI class

This also ensures that all input validation is handled by the InputValidator class

andychong23 committed:
Add save format for date and datetime objects used in tasks

andychong23 committed:
Add parseDate and parseDateTime methods

These methods ensures that string representations of accepted date and datetime formats can be parsed into their objects

andychong23 committed:
Modify implementation of EventTask and DeadLineTask to fit requirements for Level-8

For EventTask:
1. Constructor was modified to parse date and datetime inputs
2. Getters and Setters were modified to use DateTimeFormatters to get proper output for users
3. Implemented getSaveFromTime and getSaveToTime to get output for saving

For DeadLineTask:
1. Implemented getSaveDeadLine to get output for saving

andychong23 committed:
Add DateTimeFormatters that can be used by objects implementing the interfaces

andychong23 committed:
Modify dateFilePath attribute to reflect the correct path

Add format of different saves to buildTaskList method

andychong23 committed:
Merge branch 'branch-Level-8'

Main Impelmentation:
1. Keep date and datetime information as date and datetime objects instead of strings
2. Add input validation to ensure that the date information can be parsed
3. Add DateTimeFormatter objects to ensure that save and read formats are consistent
4. Modify getters for dates to ensure that they retrieve dates in the correct format
5. Move input validation from UI to InputValidator

andychong23 committed:
Merge branch 'add-gradle-support' to master for A-Gradle

andychong23 committed:
Add gradle support

Modified the properties file to use the latest version compatible with JDK 21

andychong23 committed:
Change dataFilePath to reflect data folder in project directoy

andychong23 committed:
Merge branch 'branch-A-Gradle'

Add gradle support to the project

andychong23 committed:
Add parameterized tests package for functioning of JUnit tests

andychong23 committed:
Add UnitTesting for 4 classes

1. InputValidator
2. DeadLineTask
3. EventTask
4. ToDoTask

In InputValidator, the methods isInputValid, isValidDate, isValidDateTime is tested
In DeadLineTask, the methods getDeadline, createSaveData, getTaskInformation is tested
In EventTask, the methods getDeadline, getStartDateTime, createSaveData, getTaskInformation is tested
In ToDoTask, the methods createSaveData and getTaskInformation is tested

andychong23 committed:
Merge branch 'branch-A-JUnit'

andychong23 committed:
Update .gitignore file to ignore .jar files

andychong23 committed:
Modify shadowJar to compile project into bob.jar

andychong23 committed:
Handle situations where the data folder does not exists and creates the directory

andychong23 committed:
Modify the way dataFilePath is created

Path resolve method is used instead of string concatenation

andychong23 committed:
Merge branch 'branch-A-Jar' to master

Implementation:
1. Modified .gitignore to ignore jar file in project
2. Modified build.gradle to package the app as a jar as bob.jar
3.Changed how paths are created in DataHandler and User

andychong23 committed:
Add documentation to non trivial methods

andychong23 committed:
Add underscore to separate different parts for test methods

andychong23 committed:
Modify code according to coding standards

Changed if-else indentations, try-catch indentations and do-while indentations

andychong23 committed:
Add new action to be handled

The new action's keyword is 'FIND'

andychong23 committed:
Add input validation of FIND action

Input validation for FIND checks if there are at least one argument following the FIND keyword

andychong23 committed:
Add new InputError: FIND_TOO_LITTLE_ARGUMENTS

This new enum type will return the error message that the user has provided too little arguments

andychong23 committed:
Add action processing for FIND action

andychong23 committed:
Add support for FIND action in User class

Main Implementations:
1. Add findTaskWithKeyWord method. This method takes in a keyword and searches each Task's details for a case-insensitive match
2. Overloaded getTaskList to take in a list of Tasks and output them as a String to be displayed to the user
	getTaskList has a very similar structure to the original getTaskList

andychong23 committed:
Improve error message returned for INVALID_COMMAND

Add more commands that are supported by not listed yet

andychong23 committed:
Add cheer.txt to implement CHEER action

andychong23 committed:
Modify quote such that the characters are all printable

andychong23 committed:
Implement action handling for CHEER action

Implemented a private method to get a cheer message
Added a RandomGenerator object to ActionHandler

andychong23 committed:
Implemented input validation for CHEER action

InputValidator:
1. Checked for number of arguments for CHEER action

ValidationToken:
1. Added CHEER_TOO_MANY_ARGUMENTS for InputError enum

andychong23 committed:
Implemented getting cheer.txt file from resource

andychong23 committed:
Merge branch 'branch-A-JavaDoc' with master

Add documentation to methods and classes

andychong23 committed:
Merge branch 'branch-A-CodingStandard' to master

Edit code structure such that it follows the coding standard provided

andychong23 committed:
Change implementation of findTaskwithKeyWord in User Class

Previous implementation only loops through the first n - keyWordLength
characters of taskDetail. However, it created some bugs where the keyword
could not be found. To resolve this, we loop through all the characters instead

andychong23 committed:
Merge branch 'branch-Level-9' to master

Implemented Level 9

Main Implementation:
1. ActionHandler class implments FIND command
2. InputValidator implements input validation for FIND command
3. ValidationToken implements InputError for FIND command
4. User implements findTaskWithKeyWord method for FIND command

andychong23 committed:
Merge branch 'branch-A-Cheer' to master

Fix ActionHandler and InputValidator conflict
- Include both CHEER and FIND actions

Implemention CHEER action
1. Implemented CHEER input validation in InputValidator
2. Implemented CHEER action handler in ActionHandler
3. Implemented resources package to store cheer.txt
4. Implemented cheerPath as a static variable in DataHandler
5. Implemented getCheerMessage method in ActionHandler
6. Implemented new InputError enum for CHEER

andychong23 committed:
Added comments for each unit test for each class

InputValidatorTest:
1. Added header comments to all unit tests to illustrate their purpose
2. Added a new test case for test_IsByeInputValid

DeadLineTaskTest, EventTaskTest, ToDoTaskTest:
1. Added header comments to all unit tests to illustrate their purpose

andychong23 committed:
Implmented DELETE input validation that was missed

Input validation of DELETE did not check if provided task number existed
Previous implementation only checked if a number was provided

New implementation checks if the task number exists and if the provided argument can be parsed as an integer

andychong23 committed:
Changed expected outcome for one test case in test_IsEventInputValid

andychong23 committed:
Merge branch 'branch-A-UnitTesting' into master

Main Implementations:
1. Included header comments for all unit tests
2. Handled further input validation for DELETE that was missed

andychong23 committed:
Refactor InputValidator class to split logic processing into more methods

InputValidator:
1. Move validation logic into individual methods to abstract isInputValid method
2. Carry out more checks for Event to ensure that there is at least information about the event

ValidationToken:
1. Include new InputErro to handle missing event information

andychong23 committed:
Merge branch 'branch-Refactor-InputValidator' to master

Refactor InputValidator to move validation for different commands into different methods

andychong23 committed:
Add javafx support in build.gradle and comment out checkStyle

andychong23 committed:
Add first draft for creating the GUI

andychong23 committed:
Add the entry point to the GUI

andychong23 committed:
Add default images for bob and user

andychong23 committed:
Change launcher location

andychong23 committed:
Change default user image

andychong23 committed:
Set default size for images and enable text wrapping

Implement static methods createUserDialogBox and createBobDialogBox
This allows different dialog box of different formats to be created
bobDialogBox will have its contents reversed from userDialogBox

andychong23 committed:
Move resource retrieval to BobWindow

andychong23 committed:
Implemented button clicking and enter keystroke events to trigger certain actions

Implemented setDefaultWindow to handle set up of the window for Bob GUI
Added setDefaultVBox, setDefaultScrollPane, setDefaultTextField to handle set up of
different GUI objects for BobWindow

Implemented handleEnter and handleButtonClick methods to trigger actions based on user inputs

andychong23 committed:
Implemented methods in Bob to complement the gui package

Methods impelmented:
1. New constructor that takes in a userName
2. getWelcomeMessage and getExitMessage to return text to the gui component
3. parseInput to take in the input from the gui component and evaluate whether the command is valid

andychong23 committed:
Modified UI class to fit the gui package added

Main changes:
1. Implemented new UI constructor that only takes in the botName as an input
2. Add getWelcomeMessage and getExitMessage to return welcome and exit messages to bob

andychong23 committed:
Change title of the stage

andychong23 committed:
Remove redundant import

andychong23 committed:
Implment event handling for user inputs

When user presses enter or clicks the send button, it will send the command to Bob

Other implementations:
1. Implement handleExit and handleUserLogIn to handle the entry and exit of Bob

andychong23 committed:
Remove throwing IOException

andychong23 committed:
Refactor event handling by abstracting event checking logic into their own methods

Implemented isExit and isEnter to check if the user is just sending a new command or user wants to end the chatbot
Modified handleButtonClick to handle unchecked IOException to reduce code complexity in methods that uses handleButtonClick

andychong23 committed:
Remove IOException method signatures from getWelcomeMessage and getExitMessage

andychong23 committed:
Merge branch 'branch-Level-10' to master

Main Implementation:
1. Implemented a GUI for the chatbot
2. Refactor methods to redirect input from the GUI to the correct checkers
3. Implemented new methods to redirect output to the GUI

andychong23 committed:
Refactor ActionHandler methods

Main Implementation:
1. Refactor out action processing into their own methods to increase abstraction

andychong23 committed:
Merge branch 'branch-Refactor-ActionHandler' to master

friedrice231 commented on src/main/resources/cheer.txt:
```diff
@@ -0,0 +1,5 @@
+The grass is greener on the side you water on.
+It is not a bug; it is an undocumented feature.
+There is always one more bug to fix.
+The road to success and the road to failure are almost exactly the same.
+Experience is a hard teacher because she gives the test first, the lesson afterward.
```
I like the cheers you have provided here

friedrice231 commented on src/main/java/user/User.java:
```diff
@@ -0,0 +1,174 @@
+            }
+        }
+        return stringBuilder.toString();
+    }
```
The method overloading here might be unnecessary since both implementations are almost identical. Consider refactoring the logic to reduce duplication

friedrice231 commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,158 @@
+package task;
```
I like the way you have abstracted a lot of things for the different task classes

friedrice231 commented on src/main/java/io/InputValidator.java:
```diff
@@ -0,0 +1,285 @@
+            return new ValidationToken(false, ValidationToken.InputError.MARK_INCORRECT_ARGUMENT_TYPE);
+        }
+
+        if (Integer.parseInt(userInputTokens.get(1)) < 1
+                || Integer.parseInt(userInputTokens.get(1)) > user.getNumberOfTasks()) {
+            return new ValidationToken(false, ValidationToken.InputError.INVALID_TASK_NUMBER);
+        }
```
There are quite a lot of duplicates when performing the validations in `isDeleteInputValid`, `isMarkInputValid`, `isUnmarkInputValid`. Maybe you can consider them into a reusable method to improve maintainability.

friedrice231 reviewed (commented):
Overall, I found your code easy to read and all the classes are abstracted really well. I also noted a few parts where there are duplicates, which you can consider to refactor.

andychong23 committed:
Fix bug in Bob.java

When refactoring ActionHandler, method name was changed from processEvent -> processAction
However, the change was not reflected in the method call of Bob.java
Thus, this commit rectifies this issue

andychong23 committed:
Add assertions to ensure that createdTask is not null

andychong23 committed:
Refactor getTaskList

Main Implementation:
There were 2 getTaskList method that had the same implementation with different inputs
One built the list from taskList, the other built the task list from a list provided
This commit refactors the common code into buildTaskList

Rename the original buildTaskList to buildSavedDataTaskList to be more explicit

andychong23 committed:
Merge pull request #1 from andychong23/branch-A-Assertions

Merge branch-A-Assertions to master

andychong23 committed:
Merge branch 'master' into branch-A-CodeQuality

andychong23 committed:
Merge pull request #2 from andychong23/branch-A-CodeQuality

Improve Code Quality

andychong23 committed:
Add processing for new Action: help

Main Implementation:
1. Check if the command is HELP from the user's input
2. Implemented processHelpAction as a method
3. Implemented getHelpMessage with each Action enum as the help message is associated with the action
4. Implemented static mapAction to be able to get the help message for the action command provided by the user easily

andychong23 committed:
Implemented InputValidation for help action

Main Implementations:
1. Modified isInputValid to check for help action
2. Implemented isHelpInputValid to abstract away the validation from isInputValid

andychong23 committed:
Implemented InputError enums for help action

Main Implementations:
1. HELP_TOO_MANY_ARGUMENTS and HELP_UNKNOWN_COMMANDS are potential input errors for the help action

andychong23 committed:
Merge pull request #3 from andychong23/branch-BCD-Extension

Implement help action for the user

andychong23 committed:
Modify BobWindow such that ScrollPane resizes according to scene size

Add a event listener when mouse drag is done

andychong23 committed:
Modify event for update

andychong23 committed:
Modify setDefaultWindow such that scrollPane auto fits to the screen's width and height

andychong23 committed:
Add javafx dependencies

andychong23 committed:
Ensure that jar files works after being created

There was a problem where an error was being thrown when the jar file was being run. Mainly, file not found error.

This had to do with how the cheer resource was being retrieved. During the local implementation, cheer.txt was retrieved as a resource which was converted to a URI which was then converted to a Path object.
However, in binary files, Paths to the cheer resource do not exists. Thus, the error was thrown.

Main Implementations:

build.gradle:
1. Include the javaFxVersion in dependencies

ActionHandler.java:
1. Call the new method to that retrieves the cheer messages from resources

DataHandler.java:
1. Change the handling of the cheer resource
  a. Instead of reading saving it as a file path, implemented a method getSavedCheerMessages to read the resource as a Stream and then convert it to a List to be returned

andychong23 committed:
Add documentation for getSavedCheerMessages in DataHandler.java

andychong23 committed:
Clean up the code for BobWindow class

Main Implementation:
1. Remove magic numbers and implement them as variables such as hBoxOffset, textFieldScale, buttonScale
2. Include listeners that will modify the scrollPane, hBox, textField and button widths on width changes
3. Include listeners that will modify the scrollPane height on height changes

andychong23 committed:
Added documentation to all methods in BobWindow

Removed setSize method as it has no function

andychong23 committed:
Add documentation for DialogBox class

andychong23 committed:
Add listeners that carries out certain actions when the widthProperty and heightProperty of the stage changes

andychong23 committed:
Merge pull request #4 from andychong23/branch-A-BetterGui

Branch A-Better-Gui

andychong23 committed:
Merge branch 'master' into branch-A-Release

andychong23 committed:
Remove redundant access modifiers in interfaces

andychong23 committed:
Add fix such that scroll pane is properly sized on initialization

andychong23 committed:
Add documentation to refactored methods

andychong23 committed:
Provide a full list of command when an invalid one is received

andychong23 committed:
Change naming conventions for variable declared as final

andychong23 committed:
Add padding to labels

andychong23 committed:
Make stage resizeable so that the window created from jar file is resizeable

andychong23 committed:
Add bob as an icon to the javafx window application

andychong23 committed:
Add product screenshot

andychong23 committed:
Update README user guide for Bob

andychong23 committed:
Add help command into the README

andychong23 committed:
Merge pull request #5 from andychong23/branch-A-UserGuide

Update README user guide for Bob

andychong23 committed:
Change the image src link for the Ui showcase

andychong23 committed:
Update Ui showcase image

andychong23 committed:
Merge pull request #6 from andychong23/branch-A-Release

Branch-A-Release

andychong23 committed:
Update README.md

andychong23 committed:
Update README.md

andychong23 committed:
Update README.md
